src/nvim/ex_docmd.c

```diff
@@ -1334,11 +1334,22 @@ static int parse_count(exarg_T *eap, char **errormsg, bool validate)
   // count, it's a buffer name.
   char *p;
 
-  if ((eap->argt & EX_COUNT) && ascii_isdigit(*eap->arg)
-      && (!(eap->argt & EX_BUFNAME) || *(p = skipdigits(eap->arg + 1)) == NUL
+  // Use first argument if args array exist, otherwise use eap->arg
+  char *argp = (eap->args == NULL ? eap->arg : eap->args[0]);
+
+  if ((eap->argt & EX_COUNT) && ascii_isdigit(*argp)
+      && (!(eap->argt & EX_BUFNAME) || *(p = skipdigits(argp + 1)) == NUL
           || ascii_iswhite(*p))) {
-    long n = getdigits_long(&eap->arg, false, -1);
-    eap->arg = skipwhite(eap->arg);
+    long n;
+
+    if (eap->args == NULL) {
+      n = getdigits_long(&eap->arg, false, -1);
+      eap->arg = skipwhite(eap->arg);
+    } else {
+      n = getdigits_long(&eap->arg, false, -1);
+      shift_cmd_args(eap);
```

Comment:
It is still confusing that in `vim.cmd.sleep('10m')`, the `m` is ignored and this is the same as `vim.cmd.sleep(10)`.

Reply:
Should be fixed now